legend-engine-language-pure-dsl-service-execution/src/main/java/org/finos/legend/engine/language/pure/dsl/service/execution/ServiceRunner.java

```diff
@@ -62,21 +58,13 @@ default String getCanonicalClassName()
      */
     default String run(ServiceRunnerInput serviceRunnerInput)
     {
-        try
-        {
-            PipedOutputStream outputStream = new PipedOutputStream();
-            this.run(serviceRunnerInput, outputStream);
-            PipedInputStream inputStream = new PipedInputStream(outputStream);
-            return IOUtils.toString(inputStream, Charset.defaultCharset());
-        }
-        catch (IOException e)
-        {
-            throw new RuntimeException(e);
-        }
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        this.run(serviceRunnerInput, outputStream);
+        return outputStream.toString();
```

Comment:
We should specify the charset when converting the bytes to a string. Using the default charset is always risky.

Reply:
Currently serialization is being done using the default charset and hence I am using the same conventions here. But what you said is a valid point. Definitely going forward, we need to agree on a standard charset and use it all places.